compiler/hie-core/src/Development/IDE/Core/Compile.hs

```diff
@@ -309,10 +335,15 @@ parseFileContents
        -> Maybe SB.StringBuffer -- ^ Haskell module source text (full Unicode is supported)
        -> ExceptT [FileDiagnostic] m ([FileDiagnostic], ParsedModule)
 parseFileContents preprocessor filename mbContents = do
-   contents <- liftIO $ maybe (hGetStringBuffer filename) return mbContents
    let loc  = mkRealSrcLoc (mkFastString filename) 1 1
+
+   contents <- liftIO $ maybe (hGetStringBuffer filename) return mbContents
    dflags  <- ExceptT $ parsePragmasIntoDynFlags filename contents
```

Comment:
I think you need to move this dflags line before where you runLhs

Reply:
I believe you mean move this after `runLhs`? It already is before afaict and GHC also seems to do pragma parsing afterwards which makes sense to me.